src/Compilers/CSharp/Test/Semantic/Semantics/RefEscapingTests.cs

```diff
@@ -4714,5 +4705,284 @@ .maxstack  1
+            {
+                // (9,15): error CS8352: Cannot use variable 'rs3' in this context because it may expose referenced variables outside of their declaration scope
+                //         rs2 = rs3; // 1
+                Diagnostic(ErrorCode.ERR_EscapeVariable, "rs3").WithArguments("rs3").WithLocation(9, 15)
+            };
+
+            var comp = CreateCompilation(new[] { source0, source1 }, parseOptions: TestOptions.Regular11);
+            comp.VerifyDiagnostics(source1DiagnosticsWhenSource0IsCSharp11);
+
+            comp = CreateCompilation(new[] { source0, source1 }, parseOptions: TestOptions.Regular10);
+            comp.VerifyDiagnostics();
+
+            // Reference C# 10, consume from 11
+            var comp0 = CreateCompilation(source0, parseOptions: TestOptions.Regular10);
+            comp0.VerifyDiagnostics();
+
+            var comp1 = CreateCompilation(source1, references: new[] { comp0.ToMetadataReference() }, parseOptions: TestOptions.Regular11);
+            comp1.VerifyDiagnostics();
+
+            comp1 = CreateCompilation(source1, references: new[] { comp0.EmitToImageReference() }, parseOptions: TestOptions.Regular11);
+            comp1.VerifyDiagnostics();
+
+            // Reference C# 11, consume from 10
+            comp0 = CreateCompilation(source0, parseOptions: TestOptions.Regular11);
+            comp0.VerifyDiagnostics();
+
+            comp1 = CreateCompilation(source1, references: new[] { comp0.ToMetadataReference() }, parseOptions: TestOptions.Regular10);
+            comp1.VerifyDiagnostics(source1DiagnosticsWhenSource0IsCSharp11);
+
+            comp1 = CreateCompilation(source1, references: new[] { comp0.EmitToImageReference() }, parseOptions: TestOptions.Regular10);
+            comp1.VerifyDiagnostics(source1DiagnosticsWhenSource0IsCSharp11);
+        }
+
+        [Fact]
+        public void LocalScope_DeclarationExpression_05()
+        {
+            var source = """
+                using System;
+
+                ref struct RS
+                {
+                    public RS(ref int i) => throw null!;
+                }
+
+                class Program
+                {
+                    static void M0(out RS rs1, __arglist)
+                    {
+                        // STE of __refvalue (i.e. values in __arglist) is CallingMethod.
+                        // RSTE of __refvalue is CurrentMethod.
+                        // STE of rs1 is ReturnOnly.
+                        var ai = new ArgIterator(__arglist);
+                        rs1 = __refvalue(ai.GetNextArg(), RS);
+                        rs1 = new RS(ref __refvalue(ai.GetNextArg(), int)); // 1
+                    }
+
+                    static RS M1(scoped RS rs2)
+                    {
+                        M0(out var rs3, __arglist(rs2));
+                        return rs3; // 2
+                    }
+
+                    static RS M2(scoped RS rs4)
```

Comment:
>scoped RS [](http://example.com/codeflow?start=33&length=9)

`scoped ref RS`?

Reply:
Could you please clarify why we would change this?

Reply:
`scoped RS` makes sense.